### PR TITLE
Add recipe for EiquadProg

### PR DESCRIPTION
**Pull request by jcarpent**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jcarpent** [comment]
@EtienneAr Could you confirm your interest in being a maintainer of the project?

**Comment by EtienneAr** [comment]
> @EtienneAr Could you confirm your interest in being a maintainer of the project?

I do agree.

File: recipes/eiquadprog/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+mkdir build
+cd build
+
+cmake .. \
```
[the remark this answers is not in the document]
Comment (wolfv) [reply recipes/eiquadprog/build.sh:6]: that's useful for cross-compilation

File: recipes/eiquadprog/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    - {{ pin_subpackage(name, max_pin='x.x') }}
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}  # [not win]
+    - cmake
+    - make  # [not win]
+  host:
+    - eigen
+    - boost
```
[the remark this answers is not in the document]
Comment (wolfv) [reply recipes/eiquadprog/meta.yaml:25]: you might also want to `{{ pin_compatible('boost-cpp', 'x.x') }}` for boost (I need to look up the correct pinning) if you actually depend on the library (and not just the headers?)

[the remark this answers is not in the document]
Comment (wolfv) [reply recipes/eiquadprog/meta.yaml:25]: https://github.com/conda-forge/obake-feedstock/blob/5210604b68f2e8f9cf6b4c88dd00fe4a88164dbc/recipe/meta.yaml#L43

File: recipes/eiquadprog/build.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (wolfv) [reply recipes/eiquadprog/build.sh:6]: https://github.com/conda-forge/pcre2-feedstock/blob/f1bec374c3c1c6914a97de3a484952b077950646/recipe/build.sh#L24

**Comment by jcarpent** [comment]
Many thanks @wolfv for this very quick and precise review.

**Comment by jcarpent** [comment]
Thanks a lot @wolfv for the quick merge.

**Pull request merged**